lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
I tend to agree with @Fryguy here.  Nice thing about current approach is less setup for caller for default/documented implementation (e.g. requiring self.search), other can be registered via :query_method.   Also, as noted earlier, with the find() implemented in the caller, it can be fine-tuned/performant as needed which I've done in the API Client making the queries to the Server more efficient and levering bulk queries for example.